operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java

```diff
@@ -448,6 +451,25 @@ default boolean previousAnnotationForDependentResourcesEventFiltering() {
     return true;
   }
 
+  /**
+   * For dependent resources framework can add an annotation to filter our events that are results
+   * of changes made by the framework. There are, however, few resources that do not follow the K8S
+   * API convention that changes in metadata do not increase the "metadata.generation". For these
+   * resources, the generation is increased by adding the annotation and their controller increases
+   * the observedGeneration in the status. This results in a new event, that if not handled
+   * correctly with the resource matcher yet again results in an update and a previous version
+   * annotation change, thus results in an infinite loop.
+   *
+   * <p>As a workaround, we automatically skip adding previous annotation for those well-known
+   * resources. Note that if you are sure that the matcher works (most of the cases does) for your
+   * case, you can remove the resource from the blocklist.
+   *
+   * @return blocklist of resource classes where the previous version annotation won't be used.
+   */
+  default List<Class<? extends HasMetadata>> previousAnnotationUsageBlocklist() {
```

Comment:
I would suggest also documenting the default implementation in the javadoc.

Reply:
Not sure that is needed / that helpful since users can just open the code and see the defaults.

Reply:
What would make more sense, though, is to explain what is the consequence for a resource type to be in the block list because this isn't clear at all and people might be hesitant adding a resource to the block list without more details of what happens when they do.

Reply:
Also, this should probably be a set rather than a list as the order is meaningless whereas we only want one instance of a given resource type in this collection.

Reply:
Explained the consequence and changed it to set, thank you!